util/src/main/java/com/google/cloud/hadoop/util/CredentialFactory.java

```diff
@@ -407,4 +442,26 @@ public Credential getApplicationDefaultCredentials(List<String> scopes, HttpTran
     return GoogleCredentialWithRetry.fromGoogleCredential(
         GoogleCredential.getApplicationDefault(transport, JSON_FACTORY).createScoped(scopes));
   }
+
+  // TODO (Copied (mostly) over from Google Credential since it has private scope)
+  private static PrivateKey privateKeyFromPkcs8(String privateKeyPem) throws IOException {
+    Reader reader = new StringReader(privateKeyPem);
+    Section section = PemReader.readFirstSectionAndClose(reader, "PRIVATE KEY");
+    if (section == null) {
+      throw new IOException("Invalid PKCS8 data.");
+    }
+    byte[] bytes = section.getBase64DecodedBytes();
+    PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(bytes);
+    Exception unexpectedException = null;
+    try {
+      KeyFactory keyFactory = SecurityUtils.getRsaKeyFactory();
+      PrivateKey privateKey = keyFactory.generatePrivate(keySpec);
+      return privateKey;
+    } catch (NoSuchAlgorithmException exception) {
+      unexpectedException = exception;
+    } catch (InvalidKeySpecException exception) {
+      unexpectedException = exception;
+    }
```

Comment:
Replace with: `catch (NoSuchAlgorithmException | InvalidKeySpecException e) {` and remove `unexpectedException` variable?

Reply:
Done. This was copied over from another Google library.